spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/influx/InfluxDbBatchOptionsCustomizer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.influx;
+
+import org.influxdb.BatchOptions;
+
+/**
+ * Callback interface for customizing {@code BatchOptions} beans.
+ *
+ * @author Eddú Meléndez
+ * @since 2.5.0
+ */
+@FunctionalInterface
+public interface InfluxDbBatchOptionsCustomizer {
+
+	/**
+	 * Customize the {@link BatchOptions}.
+	 * @param batchOptions the batch options to customize
+	 * @return the new batch options instance
+	 */
+	BatchOptions customize(BatchOptions batchOptions);
```

Comment:
That is an interesting idea but I am wondering why we don't provide a customizer for the high-level `InfluxDB`? If you need to customize something we might just as well give the user full control.

Reply:
There is no other properties that can be set at `influxdb` level, just `query` or `write` operations. Also, there is no easy way to fetch `BatchOptions` to override those set by configuration properties with those provided by customizers. And, due to BatchOperations provide properties that can be set via lambdas then it was a much better option to provide a customizer for it.

Reply:
> There is no other properties that can be set at influxdb level, just query or write operations.

Correct but that might change in the future and that doesn't change the fact that someone may want to customize something programmatically. Right now, you're offering that only for batch options while you could offer that for everything else. That doesn't seem legit to me.

> there is no easy way to fetch BatchOptions to override those set by configuration properties with those provided by customizers.

I know that but if you use the customizer you want to be in control. If the InfluxDB API does not offer a way to override part of the options, so be it.